### PR TITLE
Slack vitess r14.0.5 dsdefense

### PR DESCRIPTION
**Pull request by ejortegau**
## Description
This PR adds the vitess backpressure / Transaction Throttler work to our v14 release. In short, it adds the following:

* vtgates parse a comment directive specifying the query priority. If missing, a default, configurable priority is assigned to the query.
* The throttler uses the priority to determine whether a given query can or cannot be throttled.
* The throttler will throttle if the query rate is above its estimate of max query rate and there is lag.
* The throttler can now throttle write statements outside of explicit `BEGIN/COMMIT` blocks.
* The throttler is still guarded by a CLI flag so if not passed, throttling is not enabled.
* Throttled queries return a different/separate MySQL error code.

In addition, some changes were backported from newer versions to v14 for easier handling of parsing of some config options.

### CODE REVIEW

**Review by rvrangel — APPROVED** [review]
LGTM, I would get at least another approval before merging it into the branch

**Review by timvaillancourt — APPROVED** [review]

**Pull request merged**
